### PR TITLE
Create OddOrEvenChecker.java

### PR DESCRIPTION
**Pull request by srideviabhirami**
Add. Function to check if a given number is odd or even

## Description

Please describe your PR here.

## Checks

My code additions do *not* contain

- [x] `if`
- [x] `for`
- [x] `while`
- [x] `do`
- [x] `switch`
- [x] The ternary operator (`? :`)

### CODE REVIEW

File: javagony/OddOrEvenChecker.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import java.util.Scanner;
+
+public class OddOrEvenChecker {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        int number = scanner.nextInt();
+        try{
+            int remainder = 1/(number%2);
+            System.out.print(number+" is odd");
```
Comment (spenserblack) [inline javagony/OddOrEvenChecker.java]:
```suggestion
            System.out.printf("%d is odd\n", number);
```
This is more of a stylistic choice, but I think that `.printf` is more readable in this case :slightly_smiling_face:

File: javagony/OddOrEvenChecker.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        }
+        catch(ArithmeticException e){
+            System.out.print(number+" is even");
```
Comment (spenserblack) [inline javagony/OddOrEvenChecker.java]:
```suggestion
            System.out.printf("%d is even\n", number);
```

File: javagony/OddOrEvenChecker.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import java.util.Scanner;
+
+public class OddOrEvenChecker {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        int number = scanner.nextInt();
```
Comment (spenserblack) [inline javagony/OddOrEvenChecker.java]: `Scanner` can be useful, but could you use the `args` instead?
Here is an example of getting an `int` from the `args`
https://github.com/spenserblack/javagony-examples/blob/fd6b9dc49925ef7b96f4cfdfbcb15a5ffbf5cee4/javagony/Fizzbuzz.java#L4-L12

**Review by spenserblack — CHANGES_REQUESTED** [review]
Thank you for your PR! Overall it looks good :+1:
Please let me know what you think about these requested changes :slightly_smiling_face:

File: javagony/OddOrEvenChecker.java (hunk 1 above)
Comment (srideviabhirami) [reply javagony/OddOrEvenChecker.java]: I'll make the necessary changes and update it

**Review by srideviabhirami — COMMENTED** [review]
Review the changes

File: javagony/OddOrEvenChecker.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+public class OddOrEvenChecker {
+    public static void main(String[] args) {
+        int number = Integer.parseInt(args[0]);
+        try{
+            int remainder = 1/(number%2);
+            System.out.printf("%d is odd\n", number);
+        }
+        catch(ArithmeticException e){
+            System.out.printf("%d is odd\n", number);
+        }
```
Comment (spenserblack) [inline javagony/OddOrEvenChecker.java]: This change makes sure that the user has passed a command-line argument that can be parsed to an `int`.
```suggestion
        try{
            int number = Integer.parseInt(args[0]);
            int remainder = 1/(number%2);
            System.out.printf("%d is odd\n", number);
        } catch(ArrayIndexOutOfBoundsException | NumberFormatException e) {
            System.out.println("Please pass a number in the command-line");
            return;
        } catch(ArithmeticException e){
            System.out.printf("%d is odd\n", number);
        }
```

**Review by spenserblack — CHANGES_REQUESTED** [review]
Thanks! One more change and this PR should be good to go.

**Review by spenserblack — APPROVED** [review]
Thank you for your contribution!

File: javagony/OddOrEvenChecker.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    } catch(ArrayIndexOutOfBoundsException | NumberFormatException e) {
+      System.out.println("Please pass a number in the command-line");
+      return;
+    }
+
+    try {
+      int remainder = 1/(number%2);
+      System.out.printf("%d is odd\n", number);
+    } catch(ArithmeticException e) {
+      System.out.printf("%d is odd\n", number);
```
Comment (spenserblack) [inline javagony/OddOrEvenChecker.java]: Oops :laughing: 
```suggestion
      System.out.printf("%d is even\n", number);
```

**Pull request merged**